### PR TITLE
Add automatic notes generation option on release

### PR DESCRIPTION
**Pull request by WojciechBarczynski**
Adding automatic notes generation on release - https://docs.github.com/en/rest/releases/releases#create-a-release:~:text=your%20repository.%22-,generate_release_notes,-boolean

### CODE REVIEW

File: github/GitRelease.py [hunk 1]
```diff
@@ -187,7 +196,7 @@ def update_release(
         target_commitish=github.GithubObject.NotSet,
     ):
         """
-        :calls: `PATCH /repos/{owner}/{repo}/releases/{release_id} <https://docs.github.com/en/rest/reference/repos#update-a-release>`_
+        :calls: `PATCH /repo s/{owner}/{repo}/releases/{release_id} <https://docs.github.com/en/rest/reference/repos#update-a-release>`_
```
Comment (wakingrufus) [inline github/GitRelease.py:199]: is this space added intentionally?

**Comment by wakingrufus** [comment]
Hi, I was just thinking of adding this myself. Thanks for beating me to it! I hope we can get this merged.

File: github/GitRelease.py (hunk 1 above)
Comment (WojciechBarczynski) [reply github/GitRelease.py]: It's typo

**Comment by WojciechBarczynski** [comment]
> Hi, I was just thinking of adding this myself. Thanks for beating me to it! I hope we can get this merged.

@wakingrufus I'm unable to quickly fix tests, so I've added an issue (#2383 ). I'll close this PR without merging. I also hope it'll be added since we want to use it ourselfs in one project.

**Pull request closed without merging**